tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "s2n_test.h"
+
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_signature_scheme.h"
+#include "tls/s2n_signature_algorithms.h"
+
+#define LENGTH 3
+#define STUFFER_SIZE (LENGTH * TLS_SIGNATURE_SCHEME_LEN + 10)
+
+const struct s2n_signature_scheme *const test_signature_schemes[] = {
+        &s2n_ecdsa_secp384r1_sha384,
+        &s2n_rsa_pkcs1_sha256,
+        &s2n_rsa_pkcs1_sha224,
+};
+
+const struct s2n_signature_preferences test_preferences = {
+        .count = LENGTH,
+        .signature_schemes = test_signature_schemes,
+};
+
+int main(int argc, char **argv)
```

Comment:
nice additions! I think we also get good confidence in selection correctness from our integration tests which should be using the default now